searx/utils.py

```diff
@@ -371,6 +371,35 @@ def convert_str_to_int(number_str: str) -> int:
     return 0
 
 
+def extr(txt, begin, end, default=""):
```

Comment:
I'm not really sure about an empty string per default, I think that `None` would be easier to handle / more convenient

Reply:
`bool` of extracted will still be False i.e
```python
if not extracted:
    pass
```
If None is needed then pass it as default=None.

This is also tested at [https://github.com/mikf/gallery-dl/test/test_text.py#L208](https://github.com/mikf/gallery-dl/blob/ea434963aed0cfb9c9a6fd412c42631cf5091e70/test/test_text.py#L208)

Reply:
Python is a dynamically typing language, so I would also prefer the `None` data type as default.  This changes with typing.  As soon as a function has several return types, the problem of typing lies with the caller again:

![image](https://github.com/searxng/searxng/assets/554536/6721e48b-9102-4e1b-98b3-5e764424c4a3)

I would therefore prefer an empty string in the case of typing.

Since the module utils.py is a “library”, all names in it should also be typed / I have now typed the function (and slightly adapted the doc-string) ... see [diff](https://github.com/searxng/searxng/compare/bda3561d61d0a9768a0d721ab8979bde28ee801b..af61b9b6b53db92d72cd268ea339860cc946f86a)